storm-webapp/src/main/java/org/apache/storm/daemon/logviewer/webapp/LogviewerResource.java

```diff
@@ -0,0 +1,268 @@
+                             IHttpCredentialsPlugin httpCredsHandler) {
+        this.logviewer = logviewerParam;
+        this.profileHandler = profileHandler;
+        this.logDownloadHandler = logDownloadHandler;
+        this.logSearchHandler = logSearchHandler;
+        this.httpCredsHandler = httpCredsHandler;
+    }
+
+    /**
+     * Handles '/log' request.
+     */
+    @GET
+    @Path("/log")
+    public Response log(@Context HttpServletRequest request) throws IOException {
+        meterLogPageHttpRequests.mark();
+
+        try {
+            String user = httpCredsHandler.getUserName(request);
+            Integer start = request.getParameter("start") != null ? parseIntegerFromMap(request.getParameterMap(), "start") : null;
+            Integer length = request.getParameter("length") != null ? parseIntegerFromMap(request.getParameterMap(), "length") : null;
+            String decodedFileName = URLDecoder.decode(request.getParameter("file"));
+            String grep = request.getParameter("grep");
+            return logviewer.logPage(decodedFileName, start, length, grep, user);
+        } catch (InvalidRequestException e) {
+            LOG.error(e.getMessage(), e);
+            return Response.status(400).entity(e.getMessage()).build();
+        }
+    }
+
+    /**
+     * Handles '/daemonlog' request.
+     */
+    @GET
+    @Path("/daemonlog")
+    public Response daemonLog(@Context HttpServletRequest request) throws IOException {
+        meterDaemonLogPageHttpRequests.mark();
+
+        try {
+            String user = httpCredsHandler.getUserName(request);
+            Integer start = request.getParameter("start") != null ? parseIntegerFromMap(request.getParameterMap(), "start") : null;
+            Integer length = request.getParameter("length") != null ? parseIntegerFromMap(request.getParameterMap(), "length") : null;
+            String decodedFileName = URLDecoder.decode(request.getParameter("file"));
+            String grep = request.getParameter("grep");
+            return logviewer.daemonLogPage(decodedFileName, start, length, grep, user);
+        } catch (InvalidRequestException e) {
+            LOG.error(e.getMessage(), e);
+            return Response.status(400).entity(e.getMessage()).build();
+        }
+    }
+
+    /**
+     * Handles '/searchLogs' request.
+     */
+    @GET
+    @Path("/searchLogs")
+    public Response searchLogs(@Context HttpServletRequest request) throws IOException {
+        String user = httpCredsHandler.getUserName(request);
+        String topologyId = request.getParameter("topoId");
+        String portStr = request.getParameter("port");
+        String callback = request.getParameter("callback");
+        String origin = request.getHeader("Origin");
+
+        return logviewer.listLogFiles(user, portStr != null ? Integer.parseInt(portStr) : null, topologyId, callback, origin);
```

Comment:
In the original code we handled an InvalidRequestException. Do we need to do that here too?